lib/src/dart2_9_suggestors/dart2_9_utilities.dart

```diff
@@ -0,0 +1,164 @@
+// Copyright 2021 Workiva Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import 'package:analyzer/dart/ast/ast.dart';
+import 'package:analyzer/dart/ast/token.dart';
+import 'package:over_react_codemod/src/boilerplate_suggestors/boilerplate_utilities.dart';
+import 'package:over_react_codemod/src/dart2_9_suggestors/dart2_9_constants.dart';
+
+/// Returns the generated argument from [argList].
+///
+/// Used specifically to find the generated factory argument in `connect`
+/// usages.
+///
+/// Returns `null` if no generated argument found.
+///
+/// Example:
+///
+/// | input                          | output |
+/// |--------------------------------|--------|
+/// | (_$Foo)                        | _$Foo  |
+/// | (castUiFactory(_$Foo))         | _$Foo  |
+/// | (Foo)                          | null   |
+SimpleIdentifier getGeneratedFactoryArg(ArgumentList argList) {
+  final args = argList.arguments;
+  if (args.length != 1) return null;
+
+  final generatedPrefix = '_\$';
+
+  final method =
+      argList.thisOrAncestorOfType<FunctionExpressionInvocation>()?.function;
+  if (method is MethodInvocation && method.methodName.name == 'connect') {
```

Comment:
So `getGeneratedFactoryArg` is a function intended only for `connect`? Can we update the name / description to indicate that?

Reply:
Yep, you're right! I'll update that

Reply:
Updated in 17b25b1